From 1e6e4e581df661b79496fb1a182152b82990dfdb Mon Sep 17 00:00:00 2001
Message-Id: <1e6e4e581df661b79496fb1a182152b82990dfdb.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Pallavi <pallavi.g@intel.com>
Date: Tue, 4 Nov 2014 12:05:22 +0530
Subject: [PATCH 67/75] MUST_REBASE [VPG]: adf/intel/vlv: sprite src rectangle
 fixed point support

The src rectangle of the sprite plane is in 16.16 fixed point format.
Convert to integer in the flip function.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I5f32e2495bbd7399b2fbf6a5d3e45de74fa7f1e5
Signed-off-by: Pallavi <pallavi.g@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_sp_plane.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
index eb4c1af..8e82593 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
@@ -458,10 +458,10 @@ static int vlv_sp_validate(struct intel_plane *plane, struct intel_buffer *buf,
 		 * Hardware doesn't handle subpixel coordinates.
 		 * Adjust to (macro)pixel boundary
 		 */
-		config->src_x = src.x1;
-		config->src_w = (src.x2 - src.x1);
-		config->src_y = src.y1;
-		config->src_h = (src.y2 - src.y1);
+		config->src_x = src.x1 >> 16;
+		config->src_w = (src.x2 - src.x1) >> 16;
+		config->src_y = src.y1 >> 16;
+		config->src_h = (src.y2 - src.y1) >> 16;
 
 		if (format_is_yuv(buf->format)) {
 			config->src_x &= ~1;
-- 
1.7.9.5

